src/python/gudhi/wasserstein/wasserstein.py

```diff
@@ -64,17 +64,31 @@ def _build_dist_matrix(X, Y, order, internal_p):
     return Cf
 
 
-def _perstot(X, order, internal_p):
+def _perstot_autodiff(X, order, internal_p):
+    '''
+    Version of _perstot that works on eagerpy tensors.
+    '''
+    return _dist_to_diag(X, internal_p).norms.lp(order)
+
+def _perstot(X, order, internal_p, enable_autodiff):
     '''
     :param X: (n x 2) numpy.array (points of a given diagram).
     :param order: exponent for Wasserstein. Default value is 2.
     :param internal_p: Ground metric on the (upper-half) plane (i.e. norm L^p in R^2); Default value is 2 (Euclidean norm).
+    :param enable_autodiff: If X is torch.tensor, tensorflow.Tensor or jax.numpy.ndarray, make the computation
+        transparent to automatic differentiation.
+    :type enable_autodiff: bool
     :returns: float, the total persistence of the diagram (that is, its distance to the empty diagram).
     '''
-    return np.linalg.norm(_dist_to_diag(X, internal_p), ord=order)
+    if enable_autodiff:
+        import eagerpy as ep
```

Comment:
I think raising an exception is a better behavior than printing a message and returning some wrong value (the situation may be a bit different for a function that shows a graphics window and whose return objet is often ignored). ImportError also comes with a pretty clear message, it doesn't seem worth expanding on it.
On the other hand, I really should document this dependency on eagerpy. While running the code will show it, the user should have been informed earlier by the documentation.
(actually, I was thinking of bundling a fork of eagerpy with gudhi, because it is small, pure python, and I may need to patch it in places, but that's for later)